### PR TITLE
Remove duplicates for the file hashes in the SelectField

### PR DESCRIPTION
**Pull request by kvshravan**
**[FIX]**

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

---
Remove duplicates for the file hashes in the Select Field and also show got values for last 50 test runs if not null.

### CODE REVIEW

File: .pycodestylerc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [pycodestyle]
 count = True
-max-line-length = 120
+max-line-length = 122
```
Comment (canihavesomecoffee) [inline .pycodestylerc]: This shouldn't be done.

File: mod_regression/controllers.py [hunk 2]
```diff
@@ -357,18 +357,19 @@ def output_add(regression_id):
...
+            append = True
+            for output_file in result.regression_test_output.multiple_files:
+                if result.got.strip() == output_file.file_hashes.strip():
+                    append = False
+                    break
+            if append:
+                check_doubles[result.got] = int(result.test_id)
     form.output_file.choices = [(output.id, output.filename_correct + ' (original)') for output in test.output_files]
-    form.test_id.choices = [('Test id ' + str(result.test_id) + ' with output ' + result.got) for result in test_files]
+    form.test_id.choices = [('Test id ' + str(test_id) + ' with output ' + got) for got, test_id in check_doubles.items()]
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]:
```suggestion
    form.test_id.choices = [f'Test id {test_id} with output {got}' for got, test_id in check_doubles.items()]
```

Might be a better option to reduce line length a bit ;)

**Review by canihavesomecoffee — CHANGES_REQUESTED** [review]

File: mod_regression/controllers.py [hunk 3]
```diff
@@ -357,18 +357,21 @@ def output_add(regression_id):
...
+            for output_file in result.regression_test_output.multiple_files:
+                if result.got.strip() == output_file.file_hashes.strip():
+                    append = False
+                    break
+            if append:
+                check_doubles[result.got] = int(result.test_id)
     form.output_file.choices = [(output.id, output.filename_correct + ' (original)') for output in test.output_files]
-    form.test_id.choices = [('Test id ' + str(result.test_id) + ' with output ' + result.got) for result in test_files]
+    form.test_id.choices = [
+        ('Test id ' + str(test_id) + ' with output ' + got) for got, test_id in check_doubles.items()
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py:373]: Rather try to use f-strings instead, that'll shorten line length too.

Comment (kvshravan) [reply mod_regression/controllers.py:373]: Alright :)

**Comment by canihavesomecoffee** [comment]
You should add a new unittest to cover the branch, because the coverage is going down. We should aim to cover all new code with unittests :)

**Comment by kvshravan** [comment]
> You should add a new unittest to cover the branch, because the coverage is going down. We should aim to cover all new code with unittests :)

I'll try adding them tomorrow :)

**Review by thealphadollar — APPROVED** [review]
LGTM.

**Review by canihavesomecoffee — APPROVED** [review]
